src/cli.ts

```diff
@@ -9,27 +9,72 @@ import * as prompt from './prompt'
 import { JsonSchemaForTheTypeScriptCompilersConfigurationFile } from './tsconfig-schema'
 import { JsonSchemaForTheTsLintConfigurationFiles } from './tslint-schema'
 
-async function main(): Promise<void> {
-    console.log(['', 'Welcome to the Sourcegraph extension creator!', ''].join('\n'))
+interface Repository {
+    type: string
+    url: string
+    directory?: string
+}
 
-    if (!(await exists('.git'))) {
-        console.log('📘 .git directory not found, initilizing git repository')
-        await exec('git', ['init'])
+async function main(): Promise<void> {
+    async function getHTTPSGitRemoteURL(): Promise<URL | undefined> {
+        try {
+            return new URL(
+                // Simpler than using URL regex
+                exec
+                    .shellSync('git remote -v')
+                    .stdout.split('\n')[0]
+                    .split('\t')[1]
+                    .split(' ')[0]
+                    .replace(':', '/')
+                    .replace('git@', 'https://')
+                    .replace('https//', 'https:/')
```

Comment:
Instead of hand-rolling this logic, which is probably missing some cases (for example: what about `http` instead of `https`?), I would use a tested parser like https://www.npmjs.com/package/git-url-parse

Reply:
Note that git clone URLs are actually not strictly URLs, they are a superset of URLs because they also support SCP syntax. I would leave the `URL` API out for this one.